src/Service/Terminal/Service_Terminal.re

```diff
@@ -14,6 +15,14 @@ type msg =
   | ProcessTitleChanged({
       id: int,
       title: string,
+    })
+  | TerminalScreenUpdated({
+      id: int,
+      screen: [@opaque] ReveryTerminal.Screen.t,
+    })
+  | TerminalCursorMoved({
+      id: int,
+      cursor: [@opaque] ReveryTerminal.Cursor.t,
```

Comment:
Can there be other screens and cursors in this context than terminal screens and terminal cursors?

Reply:
Nope, you're right, the `Terminal` is superfluous - removed!